src/renderer/missionWindow/MissionWindow.tsx

```diff
@@ -310,10 +444,20 @@ export default class MissionWindow extends Component<ThemeProps, State> {
     return title[missionName];
   }
 
+  private unlockParameterInputs(waypointType: string): void {
+    const { locked: newLocked } = this.state;
+
+    if (waypointType === 'geofence') {
+      newLocked.geofence = false;
+    }
+
+    this.setState({ locked: newLocked });
+  }
+
   public render(): ReactNode {
+    const { checklist, locked } = this.state;
     const { theme } = this.props;
     const {
-      information,
```

Comment:
why was this deleted

Reply:
from what i remember, it gave me an error saying it was not being used.

Reply:
So I think it is because you imported a function called `readyToStart` (line 14) but the variable `readyToStart` checks for something else. I think this is why you deleted it.

I recommend you rename the function when you import it:

```ts
import { readyToStart as giveMeANameBoss } from '../../util/parameter';
```

or rename the variable:

```ts
const giveMeANameBoss = information[missionName].parameters !== undefined;
```

Reply:
update, the variable should have the new logic (if you chose to rename the variable the example is as shown):

```ts
const giveMeANameBoss = information[missionName].parameters !== undefined && readToStart(this);
```

Reply:
Lets assume for the sake of this PR that we choose to rename the variable to `unlockStartMissionButton`. It is a very clear name for the variable and what it does (unlock the start mission button so that it can be clicked).

```ts
const unlockStartMissionButton= information[missionName].parameters !== undefined;
```

So in order to update `unlockStartMissionButton` to include requiring all geofencing information to be filled out too, the variable needs to be updated as such:

```ts
const unlockStartMissionButton = information[missionName].parameters !== undefined
  && checklist.geofenceTop !== undefined
  && checklist.geofenceBottom !== undefined
  && checklist.geofenceLeft !== undefined
  && checklist.geofenceRight !== undefined;
```

## There is no need to import the `readyToStart` function from `parameter.ts`, but keep reading if you would like to know how to integrate this function too:

The `readyToStart` function's job is to check if all the values in the state variable `checklist` is not `undefined`. Then if all the values in the `checklist` is not `undefined`, it will set another state variable `ready` to true. The logic for `unlockStartMissionButton` can be updated as shown:

```ts
const { checklist, locked, ready } = this.state;

const unlockStartMissionButton = information[missionName].parameters !== undefined &&
  (ready || readyToStart(this));
```

When `readyToStart` actually returns true, the `ready` variable is updated to `true`. After that, the `unlockStartMissionButton` will never run `readyToStart` since `ready === true`.

Talking about Big-O complexity, the speed that `readyToStart` runs at is O(N). The speed after `ready` is true changes it to O(1).

Reply:
so the code without the `readyToStart` function seems to work, however if i were to use the code with the function, i would get similar errors as before.

Reply:
I dont think its a big deal with the function so feel free to leave it out. It only has 4 additional checks so 5 checks vs 1 check shouldnt be a big performance thing. Feel free to stick to the non-function one.

Im looking at the function right now myself and I dont really like it that much - I dont really like the entire mission UI class and it should be made better in the future so I'm sure that function will be removed in the short future.